decidim-admin/app/controllers/concerns/decidim/admin/needs_admin_tos_accepted.rb

```diff
@@ -37,6 +38,41 @@ def permitted_paths
       def admin_tos_path
         decidim_admin.admin_terms_show_path
       end
+
+      def user_has_any_role?
+        return true if current_user.admin
+        return true if current_user.roles.any?
+        return true if participatory_process_user_role?
+        return true if assembly_user_role?
+        return true if conference_user_role?
+        return true if voting_monitoring_commitee_member?
+
+        false
+      end
+
+      def participatory_process_user_role?
+        return false unless defined?(Decidim::ParticipatoryProcessUserRole)
```

Comment:
I am not big fan of this coupling. Maybe we could have a manifest of some type that we could use for this kind of checks?

```suggestion
        return false unless Decidim.module_installed?(:participatory_process)
```